### PR TITLE
massive refactoring bc i learned how resources work and I learned how…

### PR DESCRIPTION
**Pull request by emilyzhong**
… controllers are supposed to work but now we have more delete and update methods and also submissions for assignmetns although they dont really work

### What does this PR do?
Lots of code refactoring (splitting methods into smaller controllers)
Created delete and update methods for assignments, submissions, weeks.
Use nested resources for routes

### Status
READY

### Migrations
YES (added link to submission)

CC: @aivantg @emilyzhong @francotorium

### CODE REVIEW

File: app/views/layouts/application.html.erb [hunk 1]
```diff
@@ -6,9 +6,11 @@
...
     <%= stylesheet_link_tag    'application', media: 'all', 'data-turbolinks-track': 'reload' %>
     <%= javascript_include_tag 'application', 'data-turbolinks-track': 'reload' %>
+    <link href="https://fonts.googleapis.com/css?family=Roboto:300,400,700" rel="stylesheet">
   </head>
 
   <body>
     <%= yield %>
   </body>
 </html>
+r
```
Comment (0xFranc0) [inline app/views/layouts/application.html.erb]: nit: plz remove r

**Review by 0xFranc0 — CHANGES_REQUESTED** [review]
nit: clean layout n u gucci

File: app/views/layouts/application.html.erb (hunk 1 above)
Comment (emilyzhong) [reply app/views/layouts/application.html.erb]: rrrrrrrrrr

**Comment by emilyzhong** [comment]
mergingging!

**Pull request merged**